src/main/java/book_store/service/CategoryServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package book_store.service;
+
+import book_store.dto.category.CategoryDto;
+import book_store.exception.EntityNotFoundException;
+import book_store.mapper.CategoryMapper;
+import book_store.model.Category;
+import book_store.repository.category.CategoryRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import java.util.List;
+
+@RequiredArgsConstructor
+@Service
+public class CategoryServiceImpl implements CategoryService{
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable).stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        Category category = categoryRepository.findById(id).orElseThrow(
+                ()-> new EntityNotFoundException("Can`t find category by id" + id)
```

Comment:
```suggestion
                () -> new EntityNotFoundException("Can't find category by id: " + id)
```

don't forget to Ctrl Alt L regularly